source/simulator/helper_functions.cc

```diff
@@ -251,6 +251,68 @@ namespace aspect
   }
 
 
+  template <int dim>
+  void Simulator<dim>::generate_global_statistics()
+  {
+    // set global statistics about this time step
+    statistics.add_value("Time step number", timestep_number);
+    if (parameters.convert_to_years == true)
+      statistics.add_value("Time (years)", time / year_in_seconds);
+    else
+      statistics.add_value("Time (seconds)", time);
+
+    if (parameters.convert_to_years == true)
+      statistics.add_value("Time step size (years)", time_step / year_in_seconds);
+    else
+      statistics.add_value("Time step size (seconds)", time_step);
+
+    statistics.add_value("Number of mesh cells",
+                         triangulation.n_global_active_cells());
+
+    unsigned int n_stokes_dofs = introspection.system_dofs_per_block[0];
+    if (introspection.block_indices.velocities != introspection.block_indices.pressure)
+      n_stokes_dofs += introspection.system_dofs_per_block[introspection.block_indices.pressure];
+
+    statistics.add_value("Number of Stokes degrees of freedom", n_stokes_dofs);
+    statistics.add_value("Number of temperature degrees of freedom",
+                         introspection.system_dofs_per_block[introspection.block_indices.temperature]);
+    if (parameters.n_compositional_fields > 0)
+      statistics.add_value("Number of degrees of freedom for all compositions",
+                           parameters.n_compositional_fields
+                           * introspection.system_dofs_per_block[introspection.block_indices.compositional_fields[0]]);
+  }
+
+
+  template <int dim>
+  void Simulator<dim>::initialize_statistics()
+  {
+    // if we do not do more than one nonlinear iteration, we fill
+    // the columns with their correct value
+    if(parameters.nonlinear_solver == NonlinearSolver::IMPES || parameters.nonlinear_solver == NonlinearSolver::Advection_only)
+      generate_global_statistics();
+    // otherwise, we initialize them with zeroes
+    else
```

Comment:
I don't think I understand what you are doing here. The function is called `initialize_`, so I would have expected that it doesn't actually put anything into the statistics object. In fact, I neither understand why you put anything in there in the `if` branch, nor what the purpose of the zeros is in the `else` branch -- won't these zeros show up in the statistics file and look rather odd?